### DISCUSSION TITLE
Need a guide to fetch nft floor price prices onchain

### DISCUSSION THREAD

**Discussion by ajaythxkur** [aptos-labs/aptos-developer-discussions#461] [General]
An example will be very helpful

**Comment by 0x-j** [comment]
I think it depends on the marketplace, marketplace contract should expose a view function to return the price

here's an example with our demo marketplace 
https://github.com/aptos-labs/move-by-examples/blob/main/nft-marketplace/aptos/move/sources/marketplace.move#L191-L204

**Reply by ajaythxkur** [reply]
@0xaptosj so i need to create a marketplace contract in testnet and that will work as the oracle

**Reply by 0x-j** [reply]
nft only has a price when it's listed on an marketplace, no?

or you mean the floor price of a collection? i think you need to have some indexer that index all marketplaces that list the nft

**Reply by ajaythxkur** [reply]
Yes, i need the floor price of collection

**Reply by 0x-j** [reply]
Yeah then you need to build your own indexer I believe, there doesn't exist a public nft aggregator API

**Reply by ajaythxkur** [reply]
So how is that done, i need to go to every marketplace for apis and aggregate all of them in indexer?

**Reply by 0x-j** [reply]
Kind of, yeah it's a lot of work. But I think that's the only way.

We are indeed thinking about building an nft aggregator so we can provide a public API to get floor price, but that's still in planning.

Would be very helpful if you can share more on why you need the floor price API and what are you building

**Reply by ajaythxkur** [reply]
I was thinking of upgrading our nft loans contract.

**Reply by 0x-j** [reply]
I see, yeah I think a custom indexer is required. If you want to explore this path, can look at my full stack template which has an example indexer

**Reply by 0x-j** [reply]
https://github.com/0xaptosj/aptos-full-stack-template

**Reply by ajaythxkur** [reply]
Thanks I'll look around this example
